packages/gatsby-telemetry/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import { Telemetry, IStats } from "./telemetry"
+import { makeFlush } from "./flush"
+
+const instance = new Telemetry()
+const flush = makeFlush(instance.isTrackingEnabled())
+
+process.on(`exit`, flush)
+
+// For long running commands we want to occasionally flush the data
+// The data is also sent on exit.
+
+const interval = Number.isFinite(+process.env.TELEMETRY_BUFFER_INTERVAL!)
+  ? Math.max(Number(process.env.TELEMETRY_BUFFER_INTERVAL), 1000)
+  : 10 * 60 * 1000 // 10 min
+
+const tick = (): void => {
+  flush()
+    .catch(console.error)
+    .then(() => setTimeout(tick, interval))
+}
+
+export const trackCli = (
+  input: string | undefined,
+  tags: {} | undefined,
```

Comment:
`tags` and `opts` need to be optional, that's why the `lint` step is failing here: https://circleci.com/gh/gatsbyjs/gatsby/361624